src/app/profile/_components/ProfileHeader.tsx

```diff
@@ -4,56 +4,60 @@ import { Avatar, Button, Dialog, DialogContent, DialogTitle, IconButton, Typogra
 import PaidIcon from '@mui/icons-material/Paid';
 import AddCircleOutlineIcon from '@mui/icons-material/AddCircleOutline';
 import CloseIcon from '@mui/icons-material/Close';
+import { signOut, useSession } from "next-auth/react";
 
 type Props = {
-  username: string;
-  coinsLeft: number;
+  username: string | undefined;
```

Comment:
why undefined?

Reply:
If the user is not logged in, redirect may be a better method.

Reply:
If you get to /profile without login, username will be undefined.
You can also protect the pages with next-auth/middleware, but for now I tend to keep it simple.